query-engine/core/src/query_graph_builder/extractors/filters.rs

```diff
@@ -226,7 +226,7 @@ fn find_index_fields(name: &str, model: &ModelRef) -> QueryGraphBuilderResult<Ve
         .unique_indexes()
         .into_iter()
         .find(|index| &compound_field_name(index) == name)
-        .map(|index| index.fields())
+        .map(|index| index.scalar_fields())
```

Comment:
I don't understand this... So what on earth we are doing here? Maybe I don't understand the function, but shouldn't we just filter all fields now for the index?